src/library/searchqueryparser.cpp

```diff
@@ -88,7 +90,8 @@ void SearchQueryParser::setSearchColumns(QStringList searchColumns) {
 }
 
 QString SearchQueryParser::getTextArgument(QString argument,
-                                           QStringList* tokens) const {
+        QStringList* tokens,
+        StringMatch* matchMode) const {
```

Comment:
I dislike the implicitness of the out parameter. Eg its not fully transparent if the pointer is valid but `shouldMatchExactly` is `false`. then it'll just not set anything. A caller might expect it to always set it. The result is that if used incorrectly (which is not hard in this case), the caller could be left with an initialized variable. 

Imo the better design would be not use an out-parameter in this case and instead just return a struct with both the matchmode and the argument. 

```suggestion
struct TextArgumentResult {
    QString argumentValue;
    StringMatch matchmode;
};
TextArgumentResult SearchQueryParser::getTextArgument(QString argument,
                                           QStringList* tokens) const {
```

Reply:
> Eg its not fully transparent if the pointer is valid but shouldMatchExactly is false. then it'll just not set anything.

For now, I fixed that by setting matchMode at the beginning.

A struct would for me, too, but it requires more refactoring.

Reply:
not so much imo. PR underway

Reply:
https://github.com/ronso0/mixxx/pull/47